src_c/_sdl2/video.pyx

```diff
@@ -940,3 +974,98 @@ cdef class Renderer:
+        if res < 0:
+            raise error()
+
+    def draw_rect(self, rect):
+        # https://wiki.libsdl.org/SDL_RenderDrawRect
+        cdef SDL_Rect _rect
+        cdef SDL_Rect *rectptr = pgRect_FromObject(rect, &_rect)
+        if rectptr == NULL:
+            raise TypeError('expected a rectangle')
+        res = SDL_RenderDrawRect(self._renderer, rectptr)
+        if res < 0:
+            raise error()
+
+    def fill_rect(self, rect):
+        # https://wiki.libsdl.org/SDL_RenderFillRect
+        cdef SDL_Rect _rect
+        cdef SDL_Rect *rectptr = pgRect_FromObject(rect, &_rect)
+        if rectptr == NULL:
+            raise TypeError('expected a rectangle')
+        res = SDL_RenderFillRect(self._renderer, rectptr)
+
+        if res < 0:
+            raise error()
+
+    def to_surface(self, surface=None, area=None):
+        # https://wiki.libsdl.org/SDL_RenderReadPixels
+        """
+            Read pixels from the current rendering target and create a pygame.Surface.
+            WARNING: This is a very slow operation, and should not be used frequently.
+
+        :param surface: A surface to read the pixel data into.
+                        If ``None``, a new surface is returned.
+        :param area: The area of the screen to read pixels from.
+                     If ``None``, the entire viewport is used.
+        """
+        cdef Uint32 format
+        cdef SDL_Rect rarea
+        cdef SDL_Surface *surf
+        cdef SDL_Texture *targettex
+
+        # obtain area to use
+        if area is not None:
+            if pgRect_FromObject(area, &rarea) == NULL:
+                raise TypeError('area must be None or a rect')
+        else:
+            SDL_RenderGetViewport(self._renderer, &rarea)
+
+        # prepare surface and format
+        if surface is None:
+            # create a new surface
+            targettex = SDL_GetRenderTarget(self._renderer)
+
+            if targettex == NULL:
+                format = SDL_GetWindowPixelFormat(self._win._win)
+                if format == SDL_PIXELFORMAT_UNKNOWN:
+                    raise error()
+            else:
+                if SDL_QueryTexture(targettex, &format, NULL, NULL, NULL) < 0:
+                    raise error()
+
+            surf = SDL_CreateRGBSurfaceWithFormat(
+                0,
+                rarea.w, rarea.h,
+                SDL_BITSPERPIXEL(format),
+                format)
+            if surf == NULL:
+                raise MemoryError("not enough memory for the surface")
+
+            surface = pgSurface_New2(surf, 1)
+        elif pgSurface_Check(surface):
+            surf = pgSurface_AsSurface(surface)
+            if surf.w != rarea.w or surf.h != rarea.h:
+                raise ValueError('surface size must equal the area')
+            format = surf.format.format
+        else:
+            raise TypeError("'surface' must be a surface or None")
+
+        if SDL_RenderReadPixels(self._renderer,
```

Comment:
With [SDL_RenderReadPixels](https://wiki.libsdl.org/SDL_RenderReadPixels), I can confirm that it clips the area rect for us if it is too big.
https://github.com/SDL-mirror/SDL/blob/a9b1e5ee66144d414f342b54b6622590b730e68d/src/render/SDL_render.c#L3087

However, it returns 0 on success or a negative error code on failure. 

So it should be:
```python
if SDL_RenderReadPixels(...) < 0:
    raise error()
```